llvm/lib/Target/AArch64/AArch64TargetTransformInfo.cpp

```diff
@@ -5526,7 +5526,7 @@ InstructionCost AArch64TTIImpl::getExtendedReductionCost(
   EVT ResVT = TLI->getValueType(DL, ResTy);
 
   if (Opcode == Instruction::Add && VecVT.isSimple() && ResVT.isSimple() &&
-      VecVT.getSizeInBits() >= 64) {
+      VecVT.isFixedLengthVector() && VecVT.getSizeInBits() >= 64) {
```

Comment:
is this related to the crash in the test? Could be split off?

Reply:
I think it was to-do with a crash but the tests pass after removing this change, so I've removed it and can re-add it later on if need be.